deepchecks/vision/vision_data/format_validators.py

```diff
@@ -0,0 +1,208 @@
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module validating the VisionData functionalities implemented by the user."""
+from numbers import Number
+
+import numpy as np
+
+from deepchecks.core.errors import ValidationError
+from deepchecks.vision.vision_data import TaskType
+from deepchecks.vision.vision_data.utils import object_to_numpy, sequence_to_numpy
+
+
+def validate_images_format(images):
+    """Validate that the data is in the required format.
+
+    Parameters
+    ----------
+    images
+        The images of the batch. Result of VisionData's batch_to_images
+    Raises
+    ------
+    DeepchecksValueError
+        If the images doesn't fit the required deepchecks format.
+    """
+    try:
+        image = object_to_numpy(images[0], expected_ndim=3)
+    except TypeError as err:
+        raise ValidationError(f'The batch images must be an iterable, received {type(images)}.') from err
+    try:
+        if len(image.shape) != 3:
+            raise ValidationError('The image inside the iterable must be a 3D array.')
+    except Exception as err:
+        raise ValidationError('The image inside the iterable must be a 3D array.') from err
+    if image.shape[2] not in [1, 3]:
+        raise ValidationError('The image inside the iterable must have 1 or 3 channels.')
+    sample_min = np.min(image)
+    sample_max = np.max(image)
+    if sample_min < 0 or sample_max > 255 or sample_max <= 1:
+        raise ValidationError(f'Image data should be in uint8 format(integers between 0 and 255), '
+                              f'found values in range [{sample_min}, {sample_max}].')
+
+
+def validate_labels_format(labels, task_type: TaskType):
```

Comment:
I remember the old validations were much longer

Reply:
that is true :)